### PR TITLE
library: Add radosgw_caps to manage capabilities

### PR DESCRIPTION
**Pull request by Papawy**
This PR add `radosgw_caps` module to be able to manage RadosGW users capabilities.

Usage from the module's documentation:
```YAML
- name: add users read write and all buckets capabilities
  radosgw_caps:
    name: foo
    state: present
    caps:
      - users=read,write
      - buckets=*
- name: remove usage write capabilities
  radosgw_caps:
    name: foo
    state: absent
    caps:
      - usage=write
```

This module support check mode by simulating the original `radosgw-admin` behavior when adding capabilities.

### CODE REVIEW

File: library/radosgw_caps.py [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+    """
+    Check if we are running on a containerized cluster
+    """
+
+    if "CEPH_CONTAINER_IMAGE" in os.environ:
+        container_image = os.getenv("CEPH_CONTAINER_IMAGE")
+    else:
+        container_image = None
+
+    return container_image
```
Comment (guits) [inline library/radosgw_caps.py:182]: why don't you reuse `exec_commands()` from `./module_utils/ca_common.py` ?

File: library/radosgw_caps.py [hunk 2]
```diff
@@ -0,0 +1,428 @@
...
+
+
+def exec_commands(module, cmd):
+    """
+    Execute command(s)
+    """
+
+    rc, out, err = module.run_command(cmd)
+
+    return rc, cmd, out, err
```
Comment (guits) [inline library/radosgw_caps.py:218]: ditto

File: library/radosgw_caps.py [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+        start=str(startd),
+        end=str(endd),
+        delta=str(delta),
+        diff=diff,
+        rc=rc,
+        stdout=out.rstrip("\r\n"),
+        stderr=err.rstrip("\r\n"),
+        changed=changed,
+    )
+    module.exit_json(**result)
```
Comment (guits) [inline library/radosgw_caps.py:291]: ditto

**Review by guits — CHANGES_REQUESTED** [review]

File: library/radosgw_caps.py (hunk 3 above)
Comment (Papawy) [reply library/radosgw_caps.py:260]: The `exit_module` function from `ca_common` does not have the `diff` argument, although I could add it in `ca_common`.

Comment (guits) [reply library/radosgw_caps.py:260]: let's add it if it's possible then ? 🙂

File: library/radosgw_caps.py [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+except ImportError:
+    from module_utils.ca_common import (
+        exit_module,
+        exec_command,
+        is_containerized,
+        container_exec,
+    )
+import datetime
+import json
+import os
```
Comment (guits) [inline library/radosgw_caps.py:37]:
```suggestion
```
@Papawy flake8 is complaining about this unused module

**Review by guits — APPROVED** [review]

**Comment by guits** [comment]
jenkins test centos-non_container-all_daemons

**Comment by Papawy** [comment]
Hello is it possible to back port this PR to `stable-6.0` ?

**Pull request merged**